DataStructures/DynamicStacks/DynamicStack.java

```diff
@@ -0,0 +1,81 @@
+public class DynamicStack implements Stack{
+    //TODO: 
+    // Add two constructors: parameterless and with parameter that takes a limit
+    // If no limit is set, isFull will return false always, 
+    // otherwise it will check the current size vs the limit
+    // Add getter and setter for max nodes
+    private int numberOfNodes = 0;
+    private Integer numberOfNodesLimit = null;
+    private Node topNode = null;
+
+    public DynamicStack() {
+
+    }
+
+    public DynamicStack(Integer numberOfNodesLimit){
+        setNumberOfNodesLimit(numberOfNodesLimit);
+    }
+
+    @Override
+    public boolean push(int payload){
+        if(!isFull()){
+            Node nd = new Node(payload, topNode);
+            topNode = nd;
+            this.numberOfNodes++;
+            return true;
+        }
+        return false;        
+    } 
+
+    @Override
+    public Integer peek(){
+        if (isEmpty()){
+            return null;
+        }
+        return topNode.getPayload();
+    }
+
+    @Override
+    public Integer pop(){
+        if(isEmpty()){
+            System.out.println("No node found");
+            return null;
+        }
+        int tmpNode = topNode.getPayload();
+        topNode = topNode.getNext();
+        numberOfNodes--;
+        return tmpNode;
+    }
+
+    @Override
+    public int size(){
+        return numberOfNodes;
+    }
+
+    @Override
+    public boolean isFull(){
+        //TODO: Add logic
+        if (numberOfNodesLimit != null){
+            if (numberOfNodesLimit <= numberOfNodes) {
```

Comment:
These two can be combined on a single predicate for the if:

```suggestion
            if (numberOfNodesLimit != null && numberOfNodesLimit <= numberOfNodes) {
```
And adding a bit of convention for discussion. In some conventions, literal values, constants and known values are put on the left side of the comparisons, this has to do with the processing time of resolving variables and readability.

```suggestion
            if (null !== numberOfNodesLimit && numberOfNodes >= numberOfNodesLimit) {
```

Reply:
I've added in the suggested change, will be in the next commit.